### ISSUE TITLE
`data_dir` lacks optional build tag

### ISSUE THREAD

**Issue by tibdex** [pypa/installer#193]
This is related to #171 but this issue impacts all wheels with a build tag.

For example, using the following command to create a wheel for a package named `foo` with version 0.4.2:

```bash
python setup.py bdist_wheel --build-number 1337
```

will create a wheel named `foo-0.4.2-1337-py3-none-any.whl`. The `build_tag` in:

https://github.com/pypa/installer/blob/444e5295f2403ab303dcf14160591074396f1d21/src/installer/utils.py#L66-L68

will be `1337`.

Data files in the wheel will have paths such as `foo-0.4.2-1337.data/some_folder/some_file.txt`. Due to the implementation of `data_dir`:

https://github.com/pypa/installer/blob/444e5295f2403ab303dcf14160591074396f1d21/src/installer/sources.py#L49-L52

 we'll thus have the following variable values here:

https://github.com/pypa/installer/blob/444e5295f2403ab303dcf14160591074396f1d21/src/installer/_core.py#L43

```python
assert data_dir == "foo-0.4.2.data"
assert path == "foo-0.4.2-1337.data/some_folder/some_file.txt"
assert posixpath.commonprefix([data_dir, path]) == "foo-0.4.2"
assert posixpath.commonprefix([data_dir, path]) != data_dir
# -> the scheme is incorrectly determined as root_scheme.
```

**Comment by eli-schwartz** [comment]
This sounds like exactly the opposite of https://github.com/pypa/setuptools/issues/3997 and is probably a setuptools/wheel bug for generating a non spec-compliant artifact.

> but this issue impacts all wheels with a build tag

Presumably it doesn't include *any* wheels that possess only modules/packages and don't have data files / headers / scripts. So it would impact the minority of wheels, really.

**Comment by tibdex** [comment]
> Presumably it doesn't include any wheels that possess only modules/packages and don't have data files / headers / scripts. So it would impact the minority of wheels, really.

Yes you're right: I meant to say: "impacts all wheels with a build tag _and_ data files". I'll edit the issue description for future readers.
